src/commands/msetnx.rs

```diff
@@ -0,0 +1,204 @@
+use std::sync::{Arc, Mutex};
+
+use bytes::Bytes;
+
+use crate::commands::executable::Executable;
+use crate::commands::{CommandParser, CommandParserError};
+use crate::frame::Frame;
+use crate::store::Store;
+use crate::Error;
+
+/// Sets the given keys to their respective values.
+/// Will not perform any operation at all even if just a single key already exists.
+///
+/// Ref: <https://redis.io/docs/latest/commands/msetnx/>
```

Comment:
I think I would add more documentation like there is on https://redis.io/docs/latest/commands/msetnx/ I thinks is a bit more clear.

Reply:
I'm not sure about the atomic part, so I didn't include it. IIRC Tokio can interrupt the execution of a task and continue it after, so it could be that the operation isn't atomic as stated in the Redis docs. If that's the case, we should figure out a way to make it atomic.

Reply:
I think considering we lock/unlock before doing any operation makes this atomic. I did not know Tokio could interrupt the task. I know it would yield after on an await operation but you think it could stop before the unlock even if there are not awaits?